### ISSUE TITLE
chrome no longer provides libwidevinecdmadapter.so

### ISSUE THREAD

**Issue by pltrz** [qutebrowser/qutebrowser#4565]
As the title says, it appears that Chrome for Linux is no longer including the `libwidevinecdmadapter.so` file. It still includes the `libwidevinecdm.so` file.

How can qutebrowser load DRM content from now on?

Also, I guess the FAQ might benefit from being updated on this subject.

**Comment by jgkamat** [comment]
pltrz writes:

> As the title says, it appears that Chrome for Linux is no longer including the `libwidevinecdmadapter.so` file. It still includes the `libwidevinecdm.so` file.

from the qt5-webengine-wildvine talk page:

https://aur.archlinux.org/packages/qt5-webengine-widevine/

> According to https://bugreports.qt.io/browse/QTBUG-64071 using the latest Chrome should work with Qt 5.11.1.
> And indeed, I removed this package and https://bitmovin.com/demos/drm seems to work fine, probably using /usr/lib/chromium/libwidevinecdm.so from the chromium-widevine package (looks like libwidevinecdmadapter.so isn't needed anymore at all? Haven't tried with chromium actually being uninstalled though...).

Have you tried the instructions without libwidevinecdmadapter.so?

**Comment by pltrz** [comment]
I tried without it. It didn't appear to work for me with it in the usual location at `/usr/lib/qt5/plugins/ppapi/libwidevinecdm.so` and also at `/usr/lib/chromium/libwidevinecdm.so`. And of course for each one respectively `set qt.args '["ppapi-widevine-path=/usr/lib/.../libwidevinecdm.so"]'` and restart.

**Comment by The-Compiler** [comment]
What Qt version are you on?

**Comment by pltrz** [comment]
5.11.3

**Comment by The-Compiler** [comment]
You might need Qt 5.12 for the newest widevine, not sure.

**Comment by ba06** [comment]
Just installed qutebrowser today on Linux Mint 18. Yeah, there is no libwidevinecdmadaptor.so or qt5-webengine-widevine because I guess AUR just loves deleting widevine plugins. I've been trying to get this working for hours and nothing is working because both of those don't seem to exist online anymore: just a bunch of dead links (usually to AUR). 

Even the chromium-widevine AUR package that DOES exist has removed those plugins (as far as I can tell).

**Comment by nrdxp** [comment]
I've been trying to figure this out for a while now, as I am on NixOS and want to extend the current package by allowing a widevine option, but I haven't been able to get it working until a moment ago. After using strace I realized that webengine is just ignoring the ppapi-widevine-path arguement and searching for libwidevinecdm.so in one of a couple predefined locations.

so when I just copied (a modified) libwidevinecdm.so to /opt/google/chrome/libwidevine.so it finally worked! I dunno if they changed the argument name or just deprecated the arguement all-together. This only half way solves the problem for me as I really need to specify the path; none of these locations exists on NixOS. Just for reference according to my strace, webengine is looking for widevine in these locations.

```
/opt/google/chrome
/usr/lib/chromium
/usr/lib/chromium-browser
/usr/lib64/chromium
```
